lte/gateway/python/magma/pipelined/app/gtp_flows.py

```diff
@@ -0,0 +1,221 @@
+
+        flows.add_flow(self._datapath, match, actions=actions,
+                       priority=flows.DEFAULT_PRIORITY,
+                       table_id=self.tbl_num, goto_table=self.next_table)
+        logging.info("firstone is done")
+        # Add flow for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,in_port=OFPP_LOCAL,
+                           ipv4_dst=ue_addr , direction=Direction.IN)
+        
+        actions = [parser.OFPActionSetField(tunnel_id=0xa000128),
+                   parser.OFPActionSetField(tun_ipv4_dst=EnodeB_IP),
+                   parser.OFPActionSetField(metadata=0x7594587a00d)]
+        
+        flows.add_flow(self._datapath, match, actions=actions,
+                       priority=flows.DEFAULT_PRIORITY,
+                       table_id=self.tbl_num,goto_table=self.next_table)
+        # Add flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self.config.mtr_port,ipv4_dst=ue_addr,
+                           direction=Direction.IN)
+
+        actions = [parser.OFPActionSetField(tunnel_id=0xa000128),
+                   parser.OFPActionSetField(tun_ipv4_dst=EnodeB_IP),
+                   parser.OFPActionSetField(metadata=0x7594587a00d)]
+        
+
+        flows.add_flow(self._datapath, match, actions=actions,
+                        priority=flows.DEFAULT_PRIORITY,
+                        table_id=self.tbl_num,goto_table=self.next_table)
+
+        self.i_teid = self.i_teid + 1
+
+    def _delete_gtp_tunnel_flows(self):
+        
+        # Delete flow for gtp port
+        match = MagmaMatch(tunnel_id=self.i_teid, in_port=self.config.gtp_port)
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+        
+        # Delete flow for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,in_port=OFPP_LOCAL,
+                           ipv4_dst=self.UE_ADDR)
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+        
+        # Delete flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self.config.mtr_port,ipv4_dst=self.UE_ADDR)
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+    def _add_discard_data_gtp_tunnel_flows(self, ue_addr):
+        
+        parser = self._datapath.ofproto_parser
+        ofproto = self._datapath.ofproto
+        
+        # discard uplink Tunnel
+        match = MagmaMatch(tunnel_id=self.i_teid, in_port=self.config.gtp_port)
+        cookie = self._cookie
+        cookie_mask = self._cookie
+        flows.add_flow_data_gtp(self._datapath, match,priority=flows.DEFAULT_PRIORITY + 1,
+                                table_id=self.tbl_num,
+                                cookie=cookie, cookie_mask=cookie_mask)
+        
+        # discard downlink Tunnel for LOCAL port
+        match = MagmaMatch(in_port=OFPP_LOCAL, ipv4_dst=ue_addr)
```

Comment:
LOCAL port should not be used in flow generation. use configured uplink and downlink port names.